airflow/providers/google/cloud/links/translate.py

```diff
@@ -167,13 +167,14 @@ def persist(
         task_instance,
         model_id: str,
         project_id: str,
+        dataset_id: str | None = None,
     ):
         task_instance.xcom_push(
             context,
             key=TranslationLegacyModelPredictLink.key,
             value={
                 "location": task_instance.location,
-                "dataset_id": task_instance.model.dataset_id,
+                "dataset_id": dataset_id or task_instance.model.dataset_id,
```

Comment:
As we discussed already, I'd suggest you to not expect certain fields in the `task_instance` because it can be instance of different operators, and they are not obligated to always have a  `model` attribute.
Instead we could use only the newly added `dataset_id` field in the `TranslationLegacyModelPredictLink`, and provide it explicitly in operators. And just because they are different, it is their responsibility to find the dataset ID.
That's why, in my opinion, here we could just remove `or task_instance.model.dataset_id`, and in the `AutoMLPredictOperator` provider dataset ID using hook: `dataset_id=hook.get_model(...).dataset_id`

Reply:
Good point,  got it implemented in that way.